iceoryx2-ffi/python/README.md

```diff
@@ -1,56 +1,56 @@
 # iceoryx2-ffi-python
 
-## Running Examples
+Quick reference for commands relevant for development of the `iceoryx2` Python bindings.
+
+## Install Poetry
 
 ```sh
-export PYTHONPATH="$(git rev-parse --show-toplevel)/iceoryx2-ffi/python/python-src"
-python examples/python/event/listener.py
+curl -sSL https://install.python-poetry.org | python3 -
 ```
 
-## Setup Development Environment
+## Setup Virtual Environment
 
 ```sh
-# install maturin, see
-# https://github.com/PyO3/maturin
-
 cd $(git rev-parse --show-toplevel)
 
-# create python development environment
-python -m venv .env
-
-# enter environment
-source .env/bin/activate # or source .env/bin/activate.fish
+# Install dependencies and create virtual environment
+poetry --project iceoryx2-ffi/python install
 
-# install dependencies
-pip install pytest
-pip install prospector[with_mypy]
-pip install black
-pip install isort
-pip install bandit
+# (OPTIONAL) Enter the virtual environment - skip the 'poetry run' prefix for all commands
+eval $(poetry --project iceoryx2-ffi/python env activate)
 ```
 
 ## Development
 
 ```sh
-# compile PyO3 bindings
-cd iceoryx2-ffi/python
-maturin develop
+cd $(git rev-parse --show-toplevel)
 
-export PYTHONPATH="$(git rev-parse --show-toplevel)/iceoryx2-ffi/python/python-src"
-# test python bindings
-pytest tests/*
+# Compile PyO3 bindings
+poetry --project iceoryx2-ffi/python run maturin develop --manifest-path iceoryx2-ffi/python/Cargo.toml --target-dir target/ffi/python
 
-cd $(git rev-parse --show-toplevel)
+# Test python bindings
+poetry --project iceoryx2-ffi/python run pytest iceoryx2-ffi/python/tests/*
+
+# Run static code analysis
+poetry --project iceoryx2-ffi/python run prospector -m -D -T -s veryhigh -F --profile .prospector-iceoryx2.yaml iceoryx2-ffi/python/python-src/iceoryx2
+poetry --project iceoryx2-ffi/python run prospector -m -D -T -s veryhigh -F --profile .prospector-tests.yaml iceoryx2-ffi/python/tests
+poetry --project iceoryx2-ffi/python run prospector -m -D -T -s veryhigh -F --profile .prospector.yaml examples/python
 
-# static code analysis
-prospector -m -D -T --with-tool mypy -s veryhigh $(pwd)/examples/python
-prospector -m -D -T --with-tool mypy -s veryhigh $(pwd)/iceoryx2-ffi/python/tests
+# Run formatting: import ordering
+poetry --project iceoryx2-ffi/python run isort $(pwd)/examples/python
+poetry --project iceoryx2-ffi/python run isort $(pwd)/iceoryx2-ffi/python/python-src/iceoryx2
+poetry --project iceoryx2-ffi/python run isort $(pwd)/iceoryx2-ffi/python/tests
 
-# formatting: import ordering
-isort $(pwd)/examples/python
-isort $(pwd)/iceoryx2-ffi/python/tests
+# Run formatting
+poetry --project iceoryx2-ffi/python run black $(pwd)/examples/python
+poetry --project iceoryx2-ffi/python run black $(pwd)/iceoryx2-ffi/python/python-src/iceoryx2
+poetry --project iceoryx2-ffi/python run black $(pwd)/iceoryx2-ffi/python/tests
```

Comment:
Can you also add these commands into a shell script with the CLI options `static-code-analysis`, `import-ordering`, `formatting` and `full` ... or similar?